monitoring/terraform/stack/service.tf

```diff
@@ -0,0 +1,118 @@
+    # See https://grafana.com/docs/grafana/v9.3/setup-grafana/configure-security/configure-authentication/azuread/#enable-azure-ad-oauth-in-grafana
+    GF_AUTH_AZUREAD_NAME                       = "Azure AD"
+    GF_AUTH_AZUREAD_SCOPES                     = "openid email profile offline_access"
+    GF_AUTH_AZUREAD_ENABLED                    = true
+    GF_AUTH_AZUREAD_ALLOW_SIGN_UP              = true
+    GF_AUTH_AZUREAD_AUTO_LOGIN                 = false
+    GF_AUTH_AZUREAD_ROLE_ATTRIBUTE_STRICT      = false
+    GF_AUTH_AZUREAD_ALLOW_ASSIGN_GRAFANA_ADMIN = false
+    GF_AUTH_AZUREAD_USE_PKCE                   = true
+  }
+  grafana_secrets = {
+    GF_SECURITY_ADMIN_PASSWORD    = "monitoring/${var.namespace}/grafana/admin_password"
+    GF_AUTH_AZUREAD_CLIENT_ID     = "monitoring/${var.namespace}/grafana/azure_application_id"
+    GF_AUTH_AZUREAD_CLIENT_SECRET = "monitoring/${var.namespace}/grafana/azure_client_secret"
+    GF_AUTH_AZUREAD_AUTH_URL      = "monitoring/${var.namespace}/grafana/azure_auth_url"
+    GF_AUTH_AZUREAD_TOKEN_URL     = "monitoring/${var.namespace}/grafana/azure_token_url"
+  }
+}
+
+module "log_router_container" {
+  source = "github.com/wellcomecollection/terraform-aws-ecs-service//modules/firelens?ref=v3.13.2"
+
+  namespace                = "${var.namespace}-grafana"
+  use_privatelink_endpoint = true
+}
+
+module "log_router_container_secrets_permissions" {
+  source    = "github.com/wellcomecollection/terraform-aws-ecs-service//modules/secrets?ref=v3.13.2"
+  secrets   = module.log_router_container.shared_secrets_logging
+  role_name = module.task_definition.task_execution_role_name
+}
+
+module "grafana_app_container" {
+  source = "github.com/wellcomecollection/terraform-aws-ecs-service//modules/container_definition?ref=v3.13.2"
+
+  name  = "app"
+  image = "grafana/grafana-oss:${var.grafana_version}"
+
+  mount_points = [{
+    containerPath = "/var/lib/grafana"
+    sourceVolume  = local.efs_volume_name
+  }]
+
+  environment = local.grafana_env
+  secrets     = local.grafana_secrets
+
+  port_mappings = [{
+    containerPort = local.container_port
+    hostPort      = local.container_port
+    protocol      = "tcp"
+  }]
+
+  log_configuration = module.log_router_container.container_log_configuration
+}
+
+module "app_permissions" {
+  source    = "git::github.com/wellcomecollection/terraform-aws-ecs-service.git//modules/secrets?ref=v3.13.2"
+  secrets   = local.grafana_secrets
+  role_name = module.task_definition.task_execution_role_name
+}
+
+module "task_definition" {
+  source = "github.com/wellcomecollection/terraform-aws-ecs-service//modules/task_definition?ref=v3.13.2"
+
+  task_name    = "${var.namespace}-grafana"
+  launch_types = ["FARGATE"]
+
+  // Grafana previously ran on a t2.small (1vCPU/2GB)
+  // That was probably too large but its CPU usage is very bursty
+  // and unfortunately Fargate is not burstable (t2s are), so
+  // we're trying this as a compromise between latency and cost
```

Comment:
A couple of thoughts:

* Would it be worth running our Grafana instance on Fargate Spot to help this compromise? It can afford to be interrupted occasionally.
* Digirati have tools to spin down certain environments (e.g. staging) outside office hours. I wonder if we should consider building something similar? We'd use it in other places too, e.g. the data science APIs, staging website, maybe bits of Archivematica. (Not for this PR, but future discussion.)

Reply:
I think it's just a single boolean parameter to the service module, easy to try it and roll back later.